hw/ip/otp_ctrl/data/otp_ctrl.hjson

```diff
@@ -0,0 +1,473 @@
+  param_list: [
+    { name: "NumLcPartitionWords",
+      desc: "Number of 32bit words in the life cycle partition",
+      type: "int",
+      default: "4", // 4 x 32bit = 128bit
+      local: "true"
+    },
+    { name: "NumSecretPartitionWords",
+      desc: "Number of 32bit words in the secret partition",
+      type: "int",
+      default: "32", // 32 x 32bit = 1024bit
+      local: "true"
+    },
+    { name: "NumHwCfgWords",
+      desc: "Number of 32bit words in the hardware config partition",
+      type: "int",
+      default: "8", // 8 x 32bit = 256bit
+      local: "true"
+    },
+    // TODO: this will change once we add more switches to this partition.
+    { name: "NumHwCfgReservedRegs",
+      desc: "Number of unallocated 32bit OTP words in the hardware partition",
+      type: "int",
+      default: "6", // 6 x 32bit = 192bit
+      local: "true"
+    },
+    { name: "NumSwCfgPartitionWords",
+      desc: "Number of 32bit words in the software config partition",
+      type: "int",
+      default: "212", // 212 x 32bit = 256bit
+      local: "true"
+    },
+    { name: "NumSwCfgWindowWords",
+      desc: "Number of 32bit words in the software config partition window",
+      type: "int",
+      default: "256", // 212 x 32bit = 6784bit, but window is 256 large for alignment reasons
+      local: "true"
+    },
+    { name: "NumDebugWindowWords",
+      desc: "Number of 32bit words in the debug window.",
+      type: "int",
+      default: "500",
+      local: "true"
+    },
+  ]
+
+  /////////////////////////////
+  // Intermodule Connections //
+  /////////////////////////////
+  inter_signal_list: [
+    // Power manager init command
+    { struct: "pwr_otp_init"
+      type:   "req_rsp"
+      name:   "pwr_otp_init"
+      act:    "receiver"
+      package: "otp_ctrl_pkg"
+    }
+    // Status output to power manager
+    { struct: "otp_pwr_state"
+      type:   "broadcast"  // no `_req/rsp`
+      name:   "otp_pwr_state"
+      act:    "sender"
+      package: "otp_ctrl_pkg"
+    }
+    // LC transition command
+    { struct: "lc_otp_program"
+      type:   "req_rsp"
+      name:   "lc_otp_program"
+      act:    "receiver"
+      package: "otp_ctrl_pkg"
+    }
+    // Broadcast to LC
+    { struct: "otp_lc_data"
+      type:   "broadcast"  // no `_req/rsp`
+      name:   "otp_lc_data"
+      act:    "sender"
+      package: "otp_ctrl_pkg"
+    }
+    // Broadcast from LC
+    { struct: "lc_tx_t"
+      type:   "broadcast"  // no `_req/rsp`
+      name:   "lc_provision_en"
+      act:    "receiver"
+      package: "otp_ctrl_pkg" // TODO: move to LC package
+    }
+    { struct: "lc_tx_t"
+      type:   "broadcast"  // no `_req/rsp`
+      name:   "lc_test_en"
+      act:    "receiver"
+      package: "otp_ctrl_pkg" // TODO: move to LC package
+    }
+    //Broadcast to Key Manager
+    { struct: "keymgr_key"
+      type:   "broadcast"  // no `_req/rsp`
+      name:   "otp_keymgr_key"
+      act:    "sender"
```

Comment:
apparently it's actually requester for broadcast :)
I ran into the same issue today.

Reply:
oh ok I haven't tried to topgen this just yet. thanks for letting me know...

Reply:
Let's discuss about the naming... inter-module signal data structure is not well defined in my opinion. Clear example is `broadcast`. It is not broadcast but just one direction signal...

Reply:
so in order to make this less confusing, I think we should either use `sender/receiver`or `requester/responder` terminology, but not mix them (I am fine with both). WDYT?

Reply:
concerning the broadcast part you brought up @eunchan: are we currently able to connect multiple receivers to the same broadcast signal at the moment? I may be wrong, but I thought we can only do 1:N array connections (i.e., the sender port must have N bits or structs)...

I am asking because I think it could be helpful for certain lifecycle signals to have such a single signal broadcast option, since signals like `DFT_EN`, will have to be distributed to several modules...

Reply:
Nope. Not at this moment. I think it is necessary to clean up current inter_module signal format.